source/ceylon/future/util/sleep.ceylon

```diff
@@ -0,0 +1,18 @@
+import java.lang {
+    Thread,
+    JavaInterruptedException=InterruptedException
+}
+
+"Suspend the current thread for some amount of [[time]]."
+throws (`class InterruptedException`,
+    "If the thread is interrupted before the sleep completes")
+shared void sleep(time) {
+    "The number of milliseconds to sleep for."
+    Integer time;
+    assert (time >= 0);
+    try {
+        Thread.sleep(time);
+    } catch (JavaInterruptedException e) {
```

Comment:
I don't think so? The Ceylon exception takes the place of the Java exception, just so the user doesn't have to import java to catch it. Uncaught, it will terminate the thread just like the Java exception. It's up to the user if they want to handle it by resetting the interrupted status.